### PR TITLE
introduce a configurable delay between creating resources to prevent spurious GC

### PR DESCRIPTION
**Pull request by bparees**
fixes https://github.com/openshift/origin/issues/17085
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1509022

### CODE REVIEW

**Comment by jim-minter** [comment]
@bparees lgtm

File: pkg/templateservicebroker/servicebroker/servicebroker.go [hunk 1]
```diff
@@ -47,6 +55,13 @@ func NewBroker(saKubeClientConfig *restclient.Config, informer templateinformer.
 	configCopy := *saKubeClientConfig
 	configCopy.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: kapi.Codecs}
 
+	delay := 5
+	value := os.Getenv("TEMPLATE_SERVICE_BROKER_GC_DELAY_SECONDS")
```
Comment (ironcladlou) [inline pkg/templateservicebroker/servicebroker/servicebroker.go]: how about `TEMPLATE_SERVICE_BROKER_GC_DELAY` which is a duration string parseable with `time.ParseDuration`? More flexible, plus it looks like right now it looks like you're passing the `int` through directly as a `time.Duration` which I believe will treat your unit as nanosecond.

File: pkg/templateservicebroker/servicebroker/servicebroker.go [hunk 2]
```diff
@@ -55,6 +70,7 @@ func NewBroker(saKubeClientConfig *restclient.Config, informer templateinformer.
 		hasSynced:          informer.Informer().HasSynced,
 		templateNamespaces: templateNamespaces,
 		restmapper:         restutil.DefaultMultiRESTMapper(),
+		gcCreateDelay:      time.Duration(delay),
```
Comment (ironcladlou) [inline pkg/templateservicebroker/servicebroker/servicebroker.go]: should be `delay*time.Second` (although using time.ParseDuration might be preferable so the user can specify the unit)

Comment (bparees) [reply pkg/templateservicebroker/servicebroker/servicebroker.go]: well i'm ultimately multiplying it by time.Second at the point of use but sure i guess that would be cleaner.

Comment (ironcladlou) [reply pkg/templateservicebroker/servicebroker/servicebroker.go]: missed that, sorry

**Comment by bparees** [comment]
adopted @ironcladlou's suggestions, ptal again.

**Pull request merged**